mne/io/ctf/ctf.py

```diff
@@ -217,11 +217,11 @@ def _get_sample_info(fname, res4, system_clock):
     else:
         n_trial = n_samp // res4['nsamp']
         n_omit = n_samp_tot - n_samp
-        n_samp = n_trial * res4['nsamp']
```

Comment:
I believe the zeros get written on the disk. This also works if ``n_samps`` is set to ``n_samps_tot``, but then the data would contain zeros at the end.